### ISSUE TITLE
Explod positioning affected by pausing the game

### ISSUE THREAD

**Issue by potsmugen** [ikemen-engine/Ikemen-GO#1729]
### Describe the bug

If you pause the game, any explods will apparently be positioned one frame in the future. You can also see their position jerk a little when using frame stepping.

### To Reproduce

1. Add this code to KFM:
```
[State -2, Wood]
type = Explod
trigger1 = 1
anim = 191
postype = p1
velocity = 5, -20
accel = 0, 1
removetime = 30
```
2. Compare explod position inside and outside a pause

### Expected behavior

Paused game should have the same behavior as unpaused.

### Screenshots / Video

Normal screenshot

![ikemen012](https://github.com/ikemen-engine/Ikemen-GO/assets/107247004/9687857a-5d23-4a71-802f-ec218f715175)

Paused screenshot
![ikemen013](https://github.com/ikemen-engine/Ikemen-GO/assets/107247004/349c5866-67aa-42fd-b072-1e5cb0f560f6)

### Engine Version (or source code date)

09/04/2024 but also at least as far back as 0.98.2

### Operating system

Windows

**Comment by rakieldev** [comment]
Maybe this is related to #3080 ?

**Comment by potsmugen** [comment]
I don't think so. This one is an ancient issue. More to do with our interpolation code and different things updating at different times.

**Comment by potsmugen** [comment]
I finally managed to fix this one in my branch. PR when possible.
